ext/sodium/libsodium.stub.php

```diff
@@ -165,75 +230,116 @@ function sodium_crypto_secretstream_xchacha20poly1305_pull(string &$state, strin
 function sodium_crypto_secretstream_xchacha20poly1305_rekey(string &$state): void {}
 #endif
 
+/** @sensitive-param $key */
 function sodium_crypto_shorthash(string $message, string $key): string {}
```

Comment:
`$message`?

Reply:
This function is used to protect against attacks on hash tables, by preventing an attacker from sending too many values that hash to the same bucket, thus `$message` is not really secret: It's used as an index into that hash table.

If this would be a longer hash, then I consider this function to be similar to `hash_hmac` where the to-be-hashed data is not secret. You would generate the HMAC to verify the authenticity of the data, so this is effectively the reasoning for `sodium_crypto_sign`.